sdks/universal-router-sdk/test/uniswapTrades.test.ts

```diff
@@ -1297,6 +1400,42 @@ describe('Uniswap', () => {
         compareUniswapTrades(new UniswapTrade(buildTrade([trade]), opts), new UniswapTrade(routerTrade, opts))
       })
 
+      it('v4 - handles eth input properly', async () => {
+        const [tokenIn, tokenOut] = [Ether.onChain(1), USDC]
+        const inputAmount = ethers.utils
+          .parseUnits('1', getAmountToken(tokenIn, tokenOut, tradeType).decimals)
+          .toString()
+        const rawInputAmount = getAmount(tokenIn, tokenOut, inputAmount, tradeType)
+
+        const opts = swapOptions({})
+        const trade = await V4Trade.fromRoute(
+          new V4Route([ETH_USDC_V4], Ether.onChain(1), USDC),
+          rawInputAmount,
+          tradeType
+        )
+
+        const classicQuote: PartialClassicQuote = {
+          tokenIn: ETH_ADDRESS,
+          tokenOut: USDC.address,
+          tradeType,
+          route: [
+            [
+              // ETH_USDC_V4 pool uses ETH directly
+              mockV4PoolInRoute(
+                ETH_USDC_V4,
+                WETH, // WETH here since pools use WETH internally
```

Comment:
should be ETH, no?

Reply:
Actually, sorry it look like it does support native ETH in. Re-added the test after updating.